### ISSUE TITLE
Unhandled MobArmorEquipmentPacket ?

### ISSUE THREAD

**Issue by Misteboss** [pmmp/PocketMine-MP#521]
### Issue description
When i set an armor to a player, i have this message who is spamming the console(debug mode)

Steps to reproduce the issue
1. ... Join the server, equip an armor 
2. ... Check the console in debug mode

### OS and versions
* PocketMine-MP: last api3\network commit
* PHP: 7.0.2
* Server OS: Debian
* Game version: PE/Win10 (the two)

### Crashdump, backtrace or other files

* ... `[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Lazer
[06:30:11] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobArmorEquipmentPacket received from Misteboss1
`

**Comment by ParadiseYes** [comment]
Somehing pretty similar here:

**Comment by Misteboss** [comment]
Yes but for me it's not with the ClassicUI @ParadiseYes

**Comment by Muqsit** [comment]
Yeah, same for other packets such as PlayerFallPacket. Because these haven't been set up yet in Player.php

**Comment by dktapps** [comment]
This packet is currently not used by the PocketMine-MP core. However, the spam is a different question. Do you happen to be wearing any enchanted armour?

**Comment by Muqsit** [comment]
Reproduced. Spams every tick while the player is wearing an armor piece.

**Comment by Misteboss** [comment]
Yep it was an enchanted armor

**Comment by dktapps** [comment]
The enchanted armor spam issue is a client bug, reported as https://bugs.mojang.com/browse/MCPE-20508

**Comment by Misteboss** [comment]
Same for PlayerFallPacket (unhandled) @dktapps

**Comment by dktapps** [comment]
> This packet is currently not used by the PocketMine-MP core.

**Comment by Misteboss** [comment]
ok no problem

**Comment by Muqsit** [comment]
Um, MobEquipmentPacket too..?
```
[07:38:33] [Server thread/DEBUG]: Tried to equip Item Unknown (384:0)x1 tags:0x0a00000302005850190000000a0700646973706c61790904004c6f726508010000003300c2a772c2a76456616c756520c2a76632352058500ac2a764456e6368616e74657220c2a7664d757173697452617979616e584f0804004e616d652e00c2a772c2a76cc2a761457870657269656e636520426f74746c6520c2a772c2a73728526967687420436c69636b290000 but have Item Air (0:0)x0 in target slot
[07:38:33] [Server thread/DEBUG]: Unhandled pocketmine\network\mcpe\protocol\MobEquipmentPacket received from MuqsitRayyanXO
```

**Comment by dktapps** [comment]
Let's put it this way. There is an absolute ton of network debugging material on that branch now, so anything that PocketMine doesn't expect is going to spit out debug spam. That means most things, since PicketMine does a wide variety of things wrong. That issue is likely a desktop player pressed Q to drop an item, but PocketMine doesn't understand that yet.

**Comment by jasonwynn10** [comment]
It looks like the player tried to equip a spawn egg?

**Comment by dktapps** [comment]
@jasonwynn10 384 is bottle o' enchanting, not spawn egg.

**Comment by thebigsmileXD** [comment]
> That issue is likely a desktop player pressed Q to drop an item

It doesn't appear so. I modified pmmp drop code to allow dropping 1 item, and regardless of that change this event doesn't happen as likely as when a player moves items in their inventory or "throw" an item/ eat an item.
Accepting what the client sends seems to reduce the inventory bugging quite a bit too

**Comment by dktapps** [comment]
Closing this since it's a client issue. Incidentally this appears to be fixed in 1.2.

**Comment by TheRoyalBlock** [comment]
Hey, sorry to bump an old thread, but... 

I was trying to diagnose this error and stumbled upon this. I haven't gotten the MobArmorEquipmentPacket error, but I got theMobEquipmentPacket as it tries to equip air but there's Item ___ in target slot. As you said, @dktapps, it's a client issue, but I'd just like to let you know.
